sdcm/cluster_gce.py

```diff
@@ -104,6 +104,7 @@ def _instance_wait_safe(self, instance_method: Callable[P, R], *args: P.args, **
                 f"Timeout while running '{instance_method.__name__}' method on GCE instance '{self._instance.id}'"
             ) from None
 
+    @cache
     def _refresh_instance_state(self):
```

Comment:
IP addresses can't change in GCE ?

Reply:
As I read, only when instance is stopped which I believe we don't do and if yes, shouldn't we fix seed's in that case?

Reply:
We are doing stop only in EC2, but only since we weren't using local disks in GCP, and now we do use local disks, and should be able to do the same sequence for GCP.

If we aren't not doing it today, we shouldn't assume we are not gonna do it.

Anyhow what do you mean by seeds in this context ?

Reply:
I mean - if IP address changed, wouldn't we need to replace them in scylla.yaml?
So if we don't do it and everything works, can we assume these IP's are not changing?

Reply:
the ip address that can change is the public address, local address doesn't change in AWS (and I'm guessing in GCP as well)

anyhow you need to make sure this cache doesn't break the sequence of `refresh_ip_address` that has it's own cache removal process, you need to drop the catch on this one as well on any call to `refresh_ip_address`

hence doing this a back-end specific cache, kind of complicates the situation, i.e. that one backend has cache for this function and one doesn't

Reply:
yes, there's some issue with `refresh_ip_address`. Situation around this is very unclear - to obtain ip's we often use `_refresh_instance_state` and it's not obvious how and when to replace it with something else. There's no other unified way to get IP's. 
I'll rethink the solution.

Reply:
Messing with getting ip's is quite dangerous and a way too complicated here. I changed to safer solution. See updated commit/PR message.